File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/GoogleMap/Features/NewOverlayForm.vb
Imports Gizmox.WebGUI.Forms
Imports Gizmox.WebGUI.Common
Imports Gizmox.WebGUI.Forms.Google

Public Class NewOverlayForm
    ' Field to store overlay collection passed
    Private _overlays As GoogleMapOverlayCollection
    ' Create delegate
    Public Delegate Sub OverlayAdded()
    ' Cteate handler to store method passed to the construcor
    Public OverlayAddedHandler As OverlayAdded

    Public Sub New()
        InitializeComponent()
    End Sub

    ''' <summary>
    ''' Overloaded form constructor
    ''' </summary>
    ''' <param name="overlays">Collection of GoogleMapOverlay objects</param>
    Public Sub New(overlays As GoogleMapOverlayCollection, handler As OverlayAdded)
        InitializeComponent()
        ' Save overlays collection passed to local variable
        Me._overlays = overlays
        ' Save method passed to local variable
        Me.OverlayAddedHandler = handler
    End Sub

    ''' <summary>
    ''' Handles Click event for 'Add' button
    ''' </summary>
    Private Sub mobjAddButton_Click(sender As Object, e As EventArgs) Handles mobjAddButton.Click
        ' Set local variable for validation result
        Dim mblnIsValid As Boolean = True
        ' Set local variables for coordinates
        Dim mdblLongitude As Double = 0
        Dim mdblLatitude As Double = 0
        ' If set location by coordinates
        If mobjCoordinatesRadioButton.Checked Then
            mdblLongitude = CDbl(mobjLongitudeNUD.Value)
            mdblLatitude = CDbl(mobjLatitudeNUD.Value)
        Else
            ' If set location by name
            ' Validate location name entered
            If mobjLocationNameAddressTextBox.Text = "" Then
                mobjErrorProvider.SetError(mobjLocationNameAddressTextBox, "Please enter location name")
                mblnIsValid = False
            Else
                mobjErrorProvider.SetError(mobjLocationNameAddressTextBox, "")
            End If
        End If
        ' Validate description value entered
        If mobjDescriptionTextBox.Text = "" Then
            mobjErrorProvider.SetError(mobjDescriptionTextBox, "Please enter description")
            mblnIsValid = False
        Else
            mobjErrorProvider.SetError(mobjDescriptionTextBox, "")
        End If
        ' If all values entered are valid
        If mblnIsValid Then
            If mobjCoordinatesRadioButton.Checked Then
                ' Add new overlay item by coordinates
                Me._overlays.Add(New GoogleMapMarkerOverlay(mdblLatitude, mdblLongitude, mobjDescriptionTextBox.Text))
            Else
                ' Add new overlay item by location name
                Dim newOverlay As New GoogleMapMarkerOverlay()
                newOverlay.Address = mobjLocationNameAddressTextBox.Text
                newOverlay.WindowInfoContent = mobjDescriptionTextBox.Text
                Me._overlays.Add(newOverlay)
            End If
            ' Perform delegated method
            Me.OverlayAddedHandler()
            ' Close window
            Me.Close()
        End If
    End Sub

    ''' <summary>
    ''' Handles Click event for 'Cancel' button
    ''' </summary>
    Private Sub mobjCancelButton_Click(sender As Object, e As EventArgs) Handles mobjCancelButton.Click
        ' Close window
        Me.Close()
    End Sub

    ''' <summary>
    ''' Handles CheckedChanged event for RadioButton controls
    ''' Enables/disables input control accordind to
    ''' RadioButton option selected
    ''' </summary>
    Private Sub RadioButton_CheckedChanged(sender As Object, e As EventArgs) Handles mobjCoordinatesRadioButton.CheckedChanged, mobjLocationNameAddressRadioButton.CheckedChanged
        If mobjCoordinatesRadioButton.Checked Then
            mobjLocationNameAddressTextBox.Enabled = False
            mobjLatitudeNUD.Enabled = True
            mobjLongitudeNUD.Enabled = True
        Else
            mobjLocationNameAddressTextBox.Enabled = True
            mobjLatitudeNUD.Enabled = False
            mobjLongitudeNUD.Enabled = False
        End If
    End Sub

End Class

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/GoogleMap/Features/NewOverlayForm.cs
#region Using

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;

using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Forms;
using Gizmox.WebGUI.Forms.Google;

#endregion

namespace Gizmox.WebGUI.Forms.CompanionKit.Controls.GoogleMap.Features
{
    public partial class NewOverlayForm : Form
    {
        // Field to store overlay collection passed
        private GoogleMapOverlayCollection _overlays;
        // Create delegate
        public delegate void OverlayAdded();
        // Cteate handler to store method passed to the construcor
        public OverlayAdded OverlayAddedHandler;

        public NewOverlayForm()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Overloaded form constructor
        /// </summary>
        /// <param name="overlays">Collection of GoogleMapOverlay objects</param>
        public NewOverlayForm(GoogleMapOverlayCollection overlays, OverlayAdded handler)
        {
            InitializeComponent();
            // Save overlays collection passed to local variable
            this._overlays = overlays;
            // Save method passed to local variable
            this.OverlayAddedHandler = handler;
        }

        /// <summary>
        /// Handles Click event for 'Add' button
        /// </summary>
        private void mobjAddButton_Click(object sender, EventArgs e)
        {
            // Set local variable for validation result
            bool mblnIsValid = true;
            // Set local variables for coordinates
            double mdblLongitude = 0 ;
            double mdblLatitude = 0;
            // If set location by coordinates
            if (mobjCoordinatesRadioButton.Checked)
            {
                 mdblLongitude = (double)mobjLongitudeNUD.Value;
                 mdblLatitude = (double)mobjLatitudeNUD.Value;
            }
            // If set location by name
            else
            {
                // Validate location name entered
                if (mobjLocationNameAddressTextBox.Text == "")
                {
                    mobjErrorProvider.SetError(mobjLocationNameAddressTextBox, "Please enter location name");
                    mblnIsValid = false;
                }
                else
                {
                    mobjErrorProvider.SetError(mobjLocationNameAddressTextBox, "");
                }
            }
            // Validate description value entered
            if (mobjDescriptionTextBox.Text == "")
            {
                mobjErrorProvider.SetError(mobjDescriptionTextBox, "Please enter description");
                mblnIsValid = false;
            }
            else
            {
                mobjErrorProvider.SetError(mobjDescriptionTextBox, "");
            }
            // If all values entered are valid
            if (mblnIsValid)
            {
                if (mobjCoordinatesRadioButton.Checked)
                {
                    // Add new overlay item by coordinates
                    this._overlays.Add(new GoogleMapMarkerOverlay(mdblLatitude, mdblLongitude, mobjDescriptionTextBox.Text));
                }
                else
                {
                    // Add new overlay item by location name
                    GoogleMapMarkerOverlay newOverlay = new GoogleMapMarkerOverlay();
                    newOverlay.Address = mobjLocationNameAddressTextBox.Text;
                    newOverlay.WindowInfoContent = mobjDescriptionTextBox.Text;
                    this._overlays.Add(newOverlay);
                }
                // Perform delegated method
                this.OverlayAddedHandler();
                // Close window
                this.Close();
            }
        }

        /// <summary>
        /// Handles Click event for 'Cancel' button
        /// </summary>
        private void mobjCancelButton_Click(object sender, EventArgs e)
        {
            // Close window
            this.Close();
        }

        /// <summary>
        /// Handles CheckedChanged event for RadioButton controls
        /// Enables/disables input control accordind to
        /// RadioButton option selected
        /// </summary>
        private void RadioButton_CheckedChanged(object sender, EventArgs e)
        {
            if (mobjCoordinatesRadioButton.Checked)
            {
                mobjLocationNameAddressTextBox.Enabled = false;
                mobjLatitudeNUD.Enabled = true;
                mobjLongitudeNUD.Enabled = true;
            }
            else
            {
                mobjLocationNameAddressTextBox.Enabled = true;
                mobjLatitudeNUD.Enabled = false;
                mobjLongitudeNUD.Enabled = false;
            }
        }
    }
}